contracts/protocol/markets/basic/BasicMarket.sol

```diff
@@ -132,9 +151,21 @@ contract BasicMarket is ReentrancyGuard {
 
     /// @notice Add new prediction
     /// @param amount Amount of ForeToken
-    /// @param side Predicition side (true - positive result, false - negative result)
-    function predict(uint256 amount, bool side) external {
-        foreToken.safeTransferFrom(msg.sender, address(this), amount);
+    /// @param side Prediction side (true - positive result, false - negative result)
+    /// @param token Alternative token
+    function predict(uint256 amount, bool side, IERC20 token) external {
+        if (!tokenRegistry.isTokenEnabled(address(token))) {
+            revert("Basic Market: Token is not enabled");
+        }
+
+        uint256 predictionFee = _calculatePredictionFee(address(token), amount);
+        token.safeTransferFrom(
+            msg.sender,
+            address(this),
+            amount + predictionFee
```

Comment:
If we include the fee in the amount this will cause a wrong calculation in the `_calculatePredictionFee`. Or we add a `predictionFee` argument and calculate the `predictionFee` off chain?
```solidity
uint256 predictionFee = _calculatePredictionFee(address(token), amount);
```